view/adminhtml/web/js/import.js

```diff
@@ -42,49 +46,96 @@ define([
-                        }
-                    } else {
-                        self.updateImportingMessage("<span style='color: red;'>" + response.message + "</span>");
+
+                self.ajaxPostWithRetry(self.options.submitUrl, data, 3, function(response) {
+                    var newChunkId = chunkId + 1;
+                    switch (importStatus) {
+                        case 1:
+                            self.chunkType(newChunkId, 'customer', 'category', 2);
+                            break;
+                        case 2:
+                            self.chunkType(newChunkId, 'category', 'product', 3);
+                            break;
+                        case 3:
+                            self.chunkType(newChunkId, 'product', 'order', 4);
+                            break;
+                        case 4:
+                            if(newChunkId < self.options.orderChunks) {
+                                setTimeout(function() {
+                                    self.chunkSync(newChunkId, self.options.importStatus);
+                                }, 100);
+                            } else {
+                                self.updateImportingMessage("<span style='color: green;'>" + $t('Done! Please expect up to 30 minutes for your historical data to appear in Metrilo.') + "</span>");
+                                self.element.removeClass('disabled').addClass('success').text($t('Finished Import.'));
+                            }
+                            break;
+                        default:
+                            return false;
                     }
                 });
             },
 
+            chunkType: function(newChunkId, current, next, status) {
+                var self = this;
+                if (self.options[`${current}Chunks`] > 0) {
+                    self.options.percentage = (100 / self.options[`${current}Chunks`]);
+                }
+                if(newChunkId < self.options[`${current}Chunks`]) {
+                    setTimeout(function() {
+                        self.chunkSync(newChunkId, self.options.importStatus);
+                    }, 100);
+                } else {
+                    self.element.text($t(`Importing ${next}`));
+                    self.updateImportingMessage("<span style='color: orange;'>" + $t(`${current} import is done! Commencing ${next} import.`) + "</span>");
+                    self.options.importStatus = status;
+                    setTimeout(function() {
+                        self.chunkSync(0, self.options.importStatus);
+                    }, 2000);
+                }
+            },
+
+            ajaxPostWithRetry: function(url, data, retryCount, callback) {
+                if(retryCount) {
+                    $.post(url, data, function(response) {
+                        console.log('Successfull import of chunkData');
```

Comment:
probably don't need those

Reply:
It was used for debug purposes. Removed.